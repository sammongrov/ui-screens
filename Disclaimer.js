import React, { Component } from 'react';
import { ScrollView, View } from 'react-native';
import PropTypes from 'prop-types';
import theme, { styles } from 'react-native-theme';
import { Screen, NavBar, Button, Text } from '@ui/components';
import { translate } from '../../../src/i18n';

export default class Disclaimer extends Component {
  constructor(props) {
    super(props);
    const { header, content, accept, onAccept, locale, safeBgColors } = this.props;
    this.header = header;
    this.content = content;
    this.accept = accept;
    this.onAccept = onAccept;
    this.locale = locale;
    this.safeBgColors = safeBgColors;
  }

  componentWillMount() {
    theme.setRoot(this);
  }

  render() {
    return (
      <Screen safeBgColors={this.safeBgColors}>
        <NavBar titleText="Disclaimer" />
        <View style={[styles.flex1, styles.disclaimerContainer]}>
          <View style={styles.primaryContent}>
            <ScrollView showsVerticalScrollIndicator={false}>
              <Text style={styles.paraText}>{this.content}</Text>
            </ScrollView>
          </View>
          <Button
            testID="accept-button"
            onPress={this.onAccept}
            title={this.accept}
            buttonStyle={[styles.disclaimerButtonStyle]}
            containerStyle={[styles.paddingHorizontal80, styles.paddingVertical8]}
          />
        </View>
      </Screen>
    );
  }
}

Disclaimer.propTypes = {
  onAccept: PropTypes.func.isRequired,
  header: PropTypes.string,
  content: PropTypes.string.isRequired,
  accept: PropTypes.string,
  locale: PropTypes.string,
  safeBgColors: PropTypes.array,
};

Disclaimer.defaultProps = {
  header: translate('Disclaimer', this.locale),
  accept: translate('IAccept', this.locale),
  locale: null,
  safeBgColors: ['#000', '#000'],
};
