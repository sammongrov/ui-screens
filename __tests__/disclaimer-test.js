import React from 'react';
import renderer from 'react-test-renderer';
import { Alert } from 'react-native';
import Disclaimer from '../Disclaimer';

const onAccept = () => Alert.alert('Help');

it('Disclaimer renders correctly', () => {
  const tree = renderer
    .create(
      <Disclaimer
        onAccept={onAccept}
        header="header"
        content="content"
        accept="accept"
        locale="locale"
      />,
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});

test('name', () => {
  // array
  expect(Array.isArray(['value'])).toBe(true);
});
