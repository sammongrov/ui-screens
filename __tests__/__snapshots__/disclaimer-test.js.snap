// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Disclaimer renders correctly 1`] = `
<BVLinearGradient
  colors={
    Array [
      4278190080,
      4278190080,
    ]
  }
  locations={
    Array [
      0,
      1,
    ]
  }
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RCTSafeAreaView
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      onLayout={[Function]}
    >
      <View
        style={
          Array [
            undefined,
            null,
          ]
        }
      >
        <View
          style={
            Array [
              undefined,
              undefined,
            ]
          }
        />
        <View
          style={
            Array [
              undefined,
              undefined,
              null,
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            numberOfLines={1}
            style={
              Array [
                Object {
                  "fontFamily": "OpenSans-Regular",
                },
                Array [
                  undefined,
                  null,
                ],
              ]
            }
          >
            Disclaimer
          </Text>
        </View>
        <View
          style={
            Array [
              undefined,
              undefined,
            ]
          }
        />
      </View>
      <View
        style={
          Array [
            undefined,
            undefined,
          ]
        }
      >
        <View>
          <RCTScrollView
            showsVerticalScrollIndicator={false}
          >
            <View>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "fontFamily": "OpenSans-Regular",
                    },
                    null,
                  ]
                }
              >
                content
              </Text>
            </View>
          </RCTScrollView>
        </View>
        <View
          style={
            Array [
              Array [
                undefined,
                undefined,
              ],
              false,
            ]
          }
        >
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#2089dc",
                    "borderRadius": 3,
                    "flexDirection": "row",
                    "justifyContent": "center",
                  },
                  Array [
                    undefined,
                  ],
                  false,
                  false,
                  false,
                  undefined,
                ]
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "color": "white",
                      "fontSize": 18,
                      "padding": 8,
                      "textAlign": "center",
                    },
                    undefined,
                    false,
                    false,
                  ]
                }
              >
                accept
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTSafeAreaView>
</BVLinearGradient>
`;
